parser.py

```diff
@@ -0,0 +1,118 @@
+class StrTableParser(object):
+    """
+    Parser of string representation tables
+
+    Example of basic usage:
+    >>> StrTableParser(table=f'| head |\\n| body_1 |\\n| body_2 |').container
+    {'head': ['body_1', 'body_2']}
+    """
+
+    def __init__(self, table=None):
+        self._container = {}
+        self._table = None
+        self._fields = None
+        self._body = None
+
+        if table:
+            self.table = table
+
+    @property
+    def table(self):
+        if self._table:
+            return self._table
+        else:
+            raise ValueError('Table is empty.')
+
+    @table.setter
+    def table(self, table_in_str):
+        """
+        :param table_in_str: string representation table
+        with header in first line, columns divided by "|" and EOL "\n"
+
+        Example:
+        | header_1st_col   | header_2nd_col   | header_3rd_col   |\n
+        | row_1_of_1st_col | row_1_of_2nd_col | row_1_of_3rd_col |\n
+        | row_2_of_1st_col | row_2_of_2nd_col | row_2_of_3rd_col |
+
+        """
+        self._table = table_in_str
+
+    @staticmethod
+    def _get_values_from_str(line):
+        return [value.strip() for value in line.split('|') if value.strip()]
+
+    @staticmethod
+    def _extract_header_of_table(table):
+        """
+        >>> StrTableParser._extract_header_of_table(f'| head |\\n| body_1 |\\n| body_2 |')
+        '| head |'
+        """
+
+        return table.split("\n")[0]
+
+    @staticmethod
+    def _extract_body_of_table(table):
+        """
+        >>> StrTableParser._extract_body_of_table(f'| head |\\n| body_1 |\\n| body_2 |')
+        ['| body_1 |', '| body_2 |']
+        """
+        return table.split("\n")[1:]
+
+    def _extract_fields(self):
+        return self._get_values_from_str(self._extract_header_of_table(self.table))
+
+    @property
+    def fields(self):
```

Comment:
Lazy operations are nice when we expect a huge data set, or we have suspicions that user may not need all that heavy to process information always. I don't think this is the case here. But to me this just makes things more complex that they need to be at this point. You can always add a wrapping caching/lazy implementation latter. 

(Same applies to other similar methods)

Reply:
Let's take it offline